spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/TwoStepsConversions.java

```diff
@@ -49,11 +74,35 @@ public TwoStepsConversions(CustomConversions customConversions) {
 		this.internalConversionService = new DefaultConversionService();
 		this.customConversions = customConversions;
 		this.customConversions.registerConvertersIn(this.conversionService);
+
+		this.internalConversionService.addConverter(BYTE_ARRAY_TO_BLOB_CONVERTER);
+		this.internalConversionService.addConverter(BLOB_TO_BYTE_ARRAY_CONVERTER);
+
 	}
 
 	@Override
 	@SuppressWarnings("unchecked")
-	public <T> T convertOnRead(Object val, Class<?> targetType) {
+	public <T> T convertOnRead(Value val, Class<?> targetType, Class<?> componentType) {
+		Object unwrappedVal = val.get();
+		if (unwrappedVal instanceof List) {
```

Comment:
Here we're relying on the client library giving back `List` for all of the singular types (string, number, blob, etc) to identify if we should be converting a single item or multiple items. Though i don't expect this to change (by that i mean what if they start returning Collection or Iterable in the future), I feel like it would be safer if we got this indication of List vs single item from the POJO side. By that I mean why don't we check `targetType` to see if it is a List or single item?

Reply:
Good point. For increased robustness, I would change `List` to `Iterable` here.